### PR TITLE
[Refactor][std::string --> String] IRModule is updated with String

### PR DESCRIPTION
**Pull request by ANSHUMAN87**
Refer #5490

@tqchen , @zhiics : Please help review!

### CODE REVIEW

**Comment by jroesch** [comment]
Looks good to me, assuming all tests pass. I could use this in my next set of PRs.

**Comment by ANSHUMAN87** [comment]
@jroesch: Thanks a lot for your feedback! There is a break in python binding because of this change. I am working on it currently. Will resolve ASAP. Thanks!

File: src/printer/relay_text_printer.cc [hunk 1]
```diff
@@ -918,13 +918,29 @@ static const char* kSemVer = "v0.0.4";
...
+    ftyped = runtime::TypedPackedFunc<std::string(ObjectRef)>(
+      [&annotate](const ObjectRef& expr) -> std::string {
+        return annotate(expr);
+      });
+  }
+  doc << relay::RelayTextPrinter(show_meta_data, ftyped).PrintFinal(node);
+  return doc.str();
+}
+
+String AsTextByStr(const ObjectRef& node,
```
Comment (tqchen) [inline src/printer/relay_text_printer.cc:943]: We don't want to introduce two variants of the functions (std::string and String).

**Comment by ANSHUMAN87** [comment]
@tqchen : Your comment is handled now, Thanks!
@jroesch : The issue is resolved, test cases are passing now!

**Pull request merged**